src/librustc_typeck/check/callee.rs

```diff
@@ -23,9 +23,15 @@ use rustc_target::spec::abi;
 /// method that is called).
 pub fn check_legal_trait_for_method_call(tcx: TyCtxt<'_>, span: Span, trait_id: DefId) {
     if tcx.lang_items().drop_trait() == Some(trait_id) {
-        struct_span_err!(tcx.sess, span, E0040, "explicit use of destructor method")
-            .span_label(span, "explicit destructor calls not allowed")
-            .emit();
+        let mut err = struct_span_err!(tcx.sess, span, E0040, "explicit use of destructor method");
```

Comment:
Span here points to `Drop::drop` and I'm not sure how to get `self` argument. I have tried using `next_point`, but to no avail.

Reply:
In the call site in confirm.rs the span of the receiver is `self.self_expr.span`. In the `check/mod.rs` call site there's not even guaranteed to be an argument (such as in `let f = Drop::drop;`).

Reply:
Thanks! I have updated the pull request: now the suggestion looks much better.

Can the receiver be conditionally extracted in the `check/mod.rs` call site? In the corresponding test the following code is used:
```rust
Drop::drop(&mut Foo)
```
I think that it would be better to give a suggestion for such cases too.

Reply:
You can suggest `drop` to replace `Drop::drop` in that case.